engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/utils/DataTypesExtensions.scala

```diff
@@ -51,4 +53,13 @@ object DataTypesExtensions {
 
   }
 
+  implicit class TypingResultExtension(typingResult: TypingResult) {
```

Comment:
Let's replace with `ToDateTypeSimulator.toDataType(typingResult)` + write a comment that it doesn't really resolve `DataType` but uses our `DataTypeFactorySimulatorHolder` which simulates this resolution.